From 06ead75ab036683e07b80ddc8619eaf53b3d79b1 Mon Sep 17 00:00:00 2001
From: "Brett T. Warden" <brett.t.warden@intel.com>
Date: Thu, 20 Aug 2015 14:11:40 -0700
Subject: [PATCH] DWC_ETH_QOS: convert DWC_ETH_QOS_id to a table

pci_driver.id_table points to DWC_ETH_QOS_id, which was a single struct
instance. Converted it to a table, and added the terminating entry.

Signed-off-by: Brett T. Warden <brett.t.warden@intel.com>
---
 drivers/net/ethernet/synopsys/DWC_ETH_QOS_pci.c |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/synopsys/DWC_ETH_QOS_pci.c b/drivers/net/ethernet/synopsys/DWC_ETH_QOS_pci.c
index b3ea3a3..acd812c 100755
--- a/drivers/net/ethernet/synopsys/DWC_ETH_QOS_pci.c
+++ b/drivers/net/ethernet/synopsys/DWC_ETH_QOS_pci.c
@@ -352,15 +352,16 @@ void DWC_ETH_QOS_remove(struct pci_dev *pdev)
 	return;
 }
 
-static struct pci_device_id DWC_ETH_QOS_id = {
-	PCI_DEVICE(VENDOR_ID, DEVICE_ID)
+static struct pci_device_id DWC_ETH_QOS_id[] = {
+    { PCI_DEVICE(VENDOR_ID, DEVICE_ID) },
+    { 0, } /* terminate list */
 };
 
 struct pci_dev *DWC_ETH_QOS_pcidev;
 
 static struct pci_driver DWC_ETH_QOS_pci_driver = {
 	.name = "DWC_ETH_QOS",
-	.id_table = &DWC_ETH_QOS_id,
+	.id_table = DWC_ETH_QOS_id,
 	.probe = DWC_ETH_QOS_probe,
 	.remove = DWC_ETH_QOS_remove,
 	.shutdown = DWC_ETH_QOS_shutdown,
-- 
1.7.9.5

